compiler/ast.nim

```diff
@@ -1279,6 +1289,8 @@ proc copyType*(t: PType, owner: PSym, keepId: bool): PType =
     when debugIds: registerId(result)
   result.sym = t.sym          # backend-info should not be copied
 
+proc exactReplica*(t: PType): PType = copyType(t, t.owner, true)
```

Comment:
I think this code pre-dates the introduction of `tyAlias` (or at least my familiarity with it). I'll consider using it in the future. For now, I think our time is better spent implementing the rest of the spec in the manual and fixing all concept-related issues in the tracker.